File: backend/app/models/issue.py
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, Integer, String, Text

from app.core.database import Base


class Issue(Base):
    __tablename__ = "issues"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    category = Column(String(100), nullable=True)
    location = Column(String(200), nullable=True)
    status = Column(String(50), nullable=False, default="open")
    is_resolved = Column(Boolean, nullable=False, default=False)
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow)

File: frontend/src/types/issue.ts
export type Issue = {
  id: number;
  title: string;
  description: string;
  category?: string | null;
  location?: string | null;
  status: string;
  is_resolved: boolean;
  created_at: string;
};

export type IssueCreate = {
  title: string;
  description: string;
  category?: string | null;
  location?: string | null;
};
